### ISSUE TITLE
Default parameter value for AnyStr never accepted, yields "incompatible default ..."

### ISSUE THREAD

**Issue by shoffmeister** [python/mypy#11310]
**Bug Report**

Any attempt to specify a default parameter value for a parameter of type `AnyStr` yields a type check error `Incompatible default for argument ...`

**To Reproduce**

```python
from typing import AnyStr

def thing1(path: AnyStr = b'a bytes') -> str:
    ...

def thing2(path: AnyStr = 'a str') -> str:
    ...
```

**Expected Behavior**

No diagnostics.

**Actual Behavior**

Both(!)
-> `Incompatible default for argument "path" (default has type "bytes", argument has type "str")`
-> `Incompatible default for argument "path" (default has type "str", argument has type "bytes")`

**Your Environment**

- Mypy version used: 0.9.10

**Comment by shoffmeister** [comment]
FWIW, `pyright` does not emit any diagnostics.
